### DISCUSSION TITLE
[ideas]

### DISCUSSION THREAD

**Discussion by pcinereus** [ropensci/targets#1053] [Ideas]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Hi dev team,

As a long-time Make user I really appreciate the work you are putting into bringing these pipelines into R.  Many of the projects that I work on represent analyses of ecological monitoring data.  Although I write the code, it is the responsibility of others to run the code (every couple of months).  These users require feedback on the progress of the analyses and currently, this is usefully in the form of a command line dashboard (see screencapture below).
![image](https://user-images.githubusercontent.com/16844741/233526228-f7f9d5d6-88ef-4ce0-9a54-7c69c993635a.png)
This is achieved via the `cli` package.

I would very much like to incorporate targets pipelines into these projects as there is often a need to run the analyses multiple times (as new data trickles in within a reporting cycle).  However, currently there are two impediments:

1) in addition to the target names, it would be great if it were possible to also register a more human readable (pretty) labels associated with each target (such as the labels next to the tick marks in the screenshot above)
2) currently, `tar_make` only permits a specific set of `reporter`s, all of which would break my dashboard.  It would be great if there was a way to support user defined `reporter`s.  This way they are triggered when a target starts/fails/skips etc and have access to the meta data, yet can themselves trigger additional functions (such as writing something to a logfile and refreshing a dashboard)
3) less critically, it would also be nice to be able to group the targets into stages (for feeback purposes).  E.g. some way that the coder can indicate that a particular set of targets represent "Stage 1 - data extraction" etc.

Although I bring these issues up in the context of a command line dashboard, I have similar experience when trying to create a shiny dashboard because I still essentially need nice labels and an ability to trigger logfile and status changes.

Kind regards

Murray

**Comment by wlandau** [comment]
I like where you are going with this. Have you had a look at the dashboard in `tar_watch()`? You can include it as a module in your own apps using `tar_watch_ui()` and `tar_watch_server()`. And for dashboards, `tar_progress()` and [other progress functions](https://docs.ropensci.org/targets/reference/index.html#progress) are nicer to program on than the text output from reporters.

As for informative labels: I have resisted because they would need to go in the metadata, which needs to be super light in order to have good performance. I suggest creating informative target names in the pipeline, then adding a literate programming report with `tar_quarto()` or `tar_render()` to display and explain the most important targets.

For (3), functions like `tar_meta()` and `tar_progress()` support `tidyselect` selectors for the `names` argument, so you can do things like `tar_meta(names = contains("stage1_")`.

**Comment by pcinereus** [comment]
Great.  Thanks for those suggestions.  I will investigate a little more.
